components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/utils/RestApiPublisherUtils.java

```diff
@@ -296,4 +298,34 @@ public static File exportOperationPolicyData(OperationPolicyData policyData)
             throw new APIManagementException("Error while exporting operation policy", e);
         }
     }
+
+    /**
+     * Validates the organization considering the API endpoint URL
+     * @param endpointUrl endpoint URL relevant of the API
+     * @param requestCtxOrdId organization ID related to the API invocation
+     * @return if the endpoint belongs to an authorized organization or not
+     */
+    public static boolean isEndpointBelongingToAuthorizedOrg(String endpointUrl, String requestCtxOrdId) {
+        if (endpointUrl != null && endpointUrl.contains(APIConstants.SERVICE_ENDPOINT_URL_IDENTIFIER) &&
+                endpointUrl.split("\\.")[1] != null) {
+            String namespace = endpointUrl.split("\\.")[1];
+            if (StringUtils.isNotEmpty(namespace)) {
+                try {
+                    String orgIdFromNamespace = Rudder.getOrgIdFromNamespace(namespace);
+                    return orgIdFromNamespace.equals(requestCtxOrdId);
+                } catch (FeignException e) {
+                    if (e.status() == 404) {
+                        log.debug("Could not find an organization relevant to the given endpoint URL.");
+                    } else if (e.status() == 400){
+                        log.warn("Error occurred while validating the organization for the endpoint URL due to " +
+                                "a bad request.", e);
+                    } else if (Integer.toString(e.status()).startsWith("5")){
+                        log.warn("Error occurred while validating the organization for the endpoint URL due to " +
+                                "a server side error.", e);
+                    }
+                }
+            }
+        }
+        return false;
```

Comment:
Isn't it wrong to return false here? If we set an endpoint which is completely outside of Choreo, we should still consider as a valid endpoint.